stream.go

```diff
@@ -213,13 +213,13 @@ func newClientStream(ctx context.Context, desc *StreamDesc, cc *ClientConn, meth
 	// Provide an opportunity for the first RPC to see the first service config
 	// provided by the resolver.
 	if err := cc.waitForResolvedAddrs(ctx); err != nil {
+		cc.nameResolutionDelayed = true
```

Comment:
What is this trying to capture?  I don't think this is what you want.

For example, if name resolution takes 30 seconds to occur, and there is no deadline on the RPC, then this will not ever be set.

Also, making this a boolean field on the `ClientConn` can't be correct.

Reply:
Its for capturing the delay in name resolution https://github.com/grpc/proposal/blob/master/A72-open-telemetry-tracing.md#tracing-information

I think we need to specifically look for context timeout error to set this? waitForResolvedAddrs doesn't directly calculate delay; it waits for resolved addresses, and if resolution takes too long, it will return a timeout error?

What is the expected behavior when there is not deadline?

Reply:
The goal of the feature is to determine, if a name resolution delay occurred during the RPC's lifespan, how long that took.

So basically you want to add events to the lifecycle of the RPC when these things occur:

https://github.com/grpc/grpc-go/blob/e95782510965b12ffd97569e0709e9cb45cfb312/clientconn.go#L682-L684

Reply:
Thanks @dfawley and @purnesh42H for your thoughts, I will create a separate PR for this as a bonus feature.